invenio_rdm_records/records/systemfields/deletion_status.py

```diff
@@ -103,7 +108,7 @@ def __set__(self, record, value):
 
     def pre_commit(self, record):
         """Dump the deletion status to the record before committing."""
-        status = self._get_cache(record) or RecordDeletionStatus(None)
```

Comment:
sounds like the bug was the `RecordDeletionStatus(None)` when the record wasn't cached (yet)?

Reply:
i think so, since in the now `get_obj` method the `RecordDeletionStatus` will get the value from the model. https://github.com/inveniosoftware/invenio-rdm-records/blob/6da09ba2291a3cae84ab964bc7dde3b74cf72df1/invenio_rdm_records/records/systemfields/deletion_status.py#L98

Reply:
which makes me wonder...
i think there are some cases when there _is_ no active model backing the record, e.g. when the source for the record in question is a search result (i.e. no DB query is involved) [1]
maybe, in such cases (when the model is `None`) we should just take what the `record` dictionary says here?

[1] i've had such cases with requests recently, in my effort to create "reminder notifications" for rdm-curation reviewers.
accessing some (model-backed) systemfields would fail because the model for the `Record` object is currently `None` (and would require a DB query to set properly)

Reply:
since the tests are not green yet maybe i see this special case in the tests and have to change the code anyway. to be honest, i would like to move this code into the `get_obj` function, otherwise it would recreate some code from there to avoid to call that function